src/catkin_pkg/templates/CMakeLists.txt.in

```diff
@@ -121,6 +121,8 @@ catkin_package(
 # add_dependencies(${PROJECT_NAME} ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
 
 ## Declare a C++ executable
+## Since catkin_make collects all cmake targets it is recommended to prefix all targets with
```

Comment:
`catkin_make collects all cmake targets` is a bit too fuzzy. I would propose the following text which explains more clearly what is happening here:

> With `catkin_make` all packages are built within a single CMake context
> The recommended prefix ensures that target names across packages don't collide